File: java-client/src/main/java/io/swagger/client/api/DoNotCallComplaintsApi.java
package io.swagger.client.api;

import com.sun.jersey.api.client.GenericType;

import io.swagger.client.ApiException;
import io.swagger.client.ApiClient;
import io.swagger.client.Configuration;
import io.swagger.client.Pair;

import io.swagger.client.model.DoNotCallComplaints;

import java.util.*;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-01-20T17:12:46.004Z")
public class DoNotCallComplaintsApi {
  private ApiClient apiClient;

  public DoNotCallComplaintsApi() {
    this(Configuration.getDefaultApiClient());
  }

  public DoNotCallComplaintsApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * &lt;br /&gt;\r\n&lt;b&gt;DoNotCallComplaints&lt;/b&gt;\r\n&lt;br /&gt;Free service (with registration), providing community and government complaint lookup by phone number for up to 2,000 queries per month.  Details include number complaint rates from (FTC, FCC, IRS, Indiana Attorney  General) and key entity tag extractions from complaints.
   * This is the main funciton to get data out of the call control reporting system&lt;br /&gt;\r\n            Try with api_key &#39;demo&#39; and phone number 12674070100 (spam) 12061231234 (not spam)
   * @param phoneNumber phone number to search
   * @return DoNotCallComplaints
   */
  public DoNotCallComplaints doNotCallComplaintsDoNotCallComplaints(String phoneNumber) throws ApiException {
    Object postBody = null;
    
     // verify the required parameter 'phoneNumber' is set
     if (phoneNumber == null) {
        throw new ApiException(400, "Missing the required parameter 'phoneNumber' when calling doNotCallComplaintsDoNotCallComplaints");
     }
     
    // create path and map variables
    String path = "/api/2015-11-01/DoNotCallComplaints/{phoneNumber}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "phoneNumber" + "\\}", apiClient.escapeString(phoneNumber.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      "application/json", "text/json", "application/xml", "text/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    
    GenericType<DoNotCallComplaints> returnType = new GenericType<DoNotCallComplaints>() {};
    return apiClient.invokeAPI(path, "GET", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
}

File: javascript-client/src/api/DoNotCallComplaintsApi.js
// require files in Node.js environment
var $, DoNotCallComplaints;
if (typeof module === 'object' && module.exports) {
  $ = require('jquery');
  DoNotCallComplaints = require('../model/DoNotCallComplaints.js');
}

// export module for AMD
if ( typeof define === "function" && define.amd ) {     
	define(['jquery', 'DoNotCallComplaints'], function($, DoNotCallComplaints) {
        return DoNotCallComplaintsApi;
	 });
}

var DoNotCallComplaintsApi = function DoNotCallComplaintsApi() {
	var self = this;
  
  
  /**
   * &lt;br /&gt;\r\n&lt;b&gt;DoNotCallComplaints&lt;/b&gt;\r\n&lt;br /&gt;Free service (with registration), providing community and government complaint lookup by phone number for up to 2,000 queries per month.  Details include number complaint rates from (FTC, FCC, IRS, Indiana Attorney  General) and key entity tag extractions from complaints.
   * This is the main funciton to get data out of the call control reporting system&lt;br /&gt;\r\n            Try with api_key &#39;demo&#39; and phone number 12674070100 (spam) 12061231234 (not spam)
   * @param {String}  phoneNumber phone number to search
   * @param {function} callback the callback function
   * @return DoNotCallComplaints
   */
  self.doNotCallComplaintsDoNotCallComplaints = function(phoneNumber, callback) {
    var postBody = null;
    var postBinaryBody = null;
    
     // verify the required parameter 'phoneNumber' is set
     if (phoneNumber == null) {
        //throw new ApiException(400, "Missing the required parameter 'phoneNumber' when calling doNotCallComplaintsDoNotCallComplaints");
        var errorRequiredMsg = "Missing the required parameter 'phoneNumber' when calling doNotCallComplaintsDoNotCallComplaints";
        throw errorRequiredMsg;
     }
     
    // create path and map variables
    var basePath = 'https://www.callcontrol.com';
    // if basePath ends with a /, remove it as path starts with a leading /
    if (basePath.substring(basePath.length-1, basePath.length)=='/') {
    	basePath = basePath.substring(0, basePath.length-1);
    }
    
    var path = basePath + replaceAll(replaceAll("/api/2015-11-01/DoNotCallComplaints/{phoneNumber}", "\\{format\\}","json")
, "\\{" + "phoneNumber" + "\\}", encodeURIComponent(phoneNumber.toString()));

    var queryParams = {};
    var headerParams =  {};
    var formParams =  {};

    
    
    

    path += createQueryString(queryParams);

    var options = {type: "GET", async: true, contentType: "application/json", dataType: "json", data: postBody};
    var request = $.ajax(path, options);

    request.fail(function(jqXHR, textStatus, errorThrown){
      if (callback) {
        var error = errorThrown || textStatus || jqXHR.statusText || 'error';
        callback(null, textStatus, jqXHR, error);
      }
    });
		
    request.done(function(response, textStatus, jqXHR){
      
      /**
        * @returns DoNotCallComplaints
        */
      
      var myResponse = new DoNotCallComplaints();
      myResponse.constructFromObject(response);
      if (callback) {
        callback(myResponse, textStatus, jqXHR);
      }
      
    });
 
    return request;
  }
  
  

 	function replaceAll (haystack, needle, replace) {
		var result= haystack;
		if (needle !=null && replace!=null) {
			result= haystack.replace(new RegExp(needle, 'g'), replace);
		}
		return result;
	}

 	function createQueryString (queryParams) {
		var queryString ='';
		var i = 0;
		for (var queryParamName in queryParams) {
			if (i==0) {
				queryString += '?' ;
			} else {
				queryString += '&' ;
			}
			
			queryString +=  queryParamName + '=' + encodeURIComponent(queryParams[queryParamName]);
			i++;
		}
		
		return queryString;
	}
}

// export module for Node.js
if (typeof module === 'object' && module.exports) {
  module.exports = DoNotCallComplaintsApi;
}
